From 8ba6883b1c5b4c5f447d2921d7af3a15a898d9aa Mon Sep 17 00:00:00 2001
From: Paolo Bonzini <pbonzini@redhat.com>
Date: Tue, 22 Jul 2014 14:38:09 -0400
Subject: [virt] kvm/mmu: unify destroy_kvm_mmu with kvm_mmu_unload

Message-id: <1406040016-3289-95-git-send-email-pbonzini@redhat.com>
Patchwork-id: 86173
O-Subject: [RHEL7 PATCH v2 094/221] KVM: mmu: unify destroy_kvm_mmu with kvm_mmu_unload
Bugzilla: 1116936
RH-Acked-by: Radim Krcmar <rkrcmar@redhat.com>
RH-Acked-by: Bandan Das <bsd@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

They do the same thing, and destroy_kvm_mmu can be confused with
kvm_mmu_destroy.

Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
Signed-off-by: Gleb Natapov <gleb@redhat.com>
(cherry picked from commit 95f93af4add9047f7d1ae4f43d12e66f2ba2d6f0)
Signed-off-by: Paolo Bonzini <pbonzini@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kvm/mmu.c b/arch/x86/kvm/mmu.c
index 07b31b7..65d4909 100644
--- a/arch/x86/kvm/mmu.c
+++ b/arch/x86/kvm/mmu.c
@@ -3870,18 +3870,11 @@ static int init_kvm_mmu(struct kvm_vcpu *vcpu)
   return init_kvm_softmmu(vcpu);
 }
 
-static void destroy_kvm_mmu(struct kvm_vcpu *vcpu)
+int kvm_mmu_reset_context(struct kvm_vcpu *vcpu)
 {
  ASSERT(vcpu);
- if (VALID_PAGE(vcpu->arch.mmu.root_hpa)) {
-  mmu_free_roots(vcpu);
-  WARN_ON(VALID_PAGE(vcpu->arch.mmu.root_hpa));
- }
-}
 
-int kvm_mmu_reset_context(struct kvm_vcpu *vcpu)
-{
- destroy_kvm_mmu(vcpu);
+ kvm_mmu_unload(vcpu);
  return init_kvm_mmu(vcpu);
 }
 EXPORT_SYMBOL_GPL(kvm_mmu_reset_context);
@@ -3907,6 +3900,7 @@ EXPORT_SYMBOL_GPL(kvm_mmu_load);
 void kvm_mmu_unload(struct kvm_vcpu *vcpu)
 {
  mmu_free_roots(vcpu);
+ WARN_ON(VALID_PAGE(vcpu->arch.mmu.root_hpa));
 }
 EXPORT_SYMBOL_GPL(kvm_mmu_unload);
 
@@ -4550,7 +4544,7 @@ void kvm_mmu_destroy(struct kvm_vcpu *vcpu)
 {
  ASSERT(vcpu);
 
- destroy_kvm_mmu(vcpu);
+ kvm_mmu_unload(vcpu);
  free_mmu_pages(vcpu);
  mmu_free_memory_caches(vcpu);
 }
-- 
1.7.1